app/models/administrator.rb

```diff
@@ -1,4 +1,12 @@
 class Administrator < ApplicationRecord
+  def self.authenticate(username, password)
+    admin = Administrator.find_by(username: username)
+    if admin.nil?
```

Comment:
We're ending up querying for the administrator twice:  once here, and once in the controller that uses it.

Most places I've seen this kind of accessor, it returns the found administrator, and the controller uses the result, rather than forcing a secondary lookup.